packages/interface/src/store/actions.js

```diff
@@ -27,9 +56,15 @@ export const enableComplementaryArea = ( scope, area ) => ( { registry } ) => {
  * @param {string} scope Complementary area scope.
  */
 export const disableComplementaryArea = ( scope ) => ( { registry } ) => {
-	registry
-		.dispatch( preferencesStore )
-		.set( scope, 'complementaryArea', null );
+	const isComplementaryAreaVisible = registry
+		.select( preferencesStore )
+		.get( scope, 'isComplementaryAreaVisible' );
+
+	if ( isComplementaryAreaVisible ) {
+		registry
+			.dispatch( preferencesStore )
+			.set( scope, 'isComplementaryAreaVisible', false );
+	}
```

Comment:
I see that reducer has the `DISABLE_COMPLEMENTARY_AREA` action case, but it's never dispatched here. Is this on purpose?

Reply:
Oh, good catch, that was left behind from an earlier iteration. It isn't needed in the reducer so I'll remove it.

Reasoning - if a specific sidebar is opened, `enableComplementaryArea` always overwrites the `area`, so removing it isn't needed. Even if some code decides to show the sidebar by toggling the `isComplementaryAreaVisible` preference, it's probably good to keep the last active `area` in state so that it's the one shown.

Reply:
Thanks for the explanation, Dan.